code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/internal/EdgePersonalizationResponseHandler.java

```diff
@@ -119,10 +131,11 @@ class EdgePersonalizationResponseHandler {
                     propositions.addAll(propositionList);
                 }
 
-                final Map<InboundType, List<LaunchRule>> parsedRules = parsePropositions(propositions, surfaces, false, false);
-                final List<LaunchRule> inAppRules = parsedRules.get(InboundType.INAPP);
+                final ParsedPropositions parsedPropositions = new ParsedPropositions(cachedPropositions, surfaces, extensionApi);
+                final Map<Surface, List<LaunchRule>> inAppRules = parsedPropositions.surfaceRulesByInboundType.get(InboundType.INAPP);
                 // register any in-app propositions which were previously cached
-                launchRulesEngine.replaceRules(inAppRules);
+                final List<LaunchRule> rulesToReplace = inAppRules.get(new Surface());
```

Comment:
Please add null check in case there are no inAppRuless

Reply:
the replaceRules method will null check the passed in rule list prior to attempting to replace rules